flow/aggregates.py

```diff
@@ -469,15 +467,28 @@ def __init__(self, aggregator_function, sort_by, reverse_order, select):
         self._select = select
 
     def __call__(self, project):
-        jobs = list(project)
-        if self._select is not None:
-            jobs = list(filter(self._select, jobs))
-        if self._sort_by is not None:
-            jobs = sorted(jobs,
+        if self._sort_by is None:
+            jobs = list(project)
+        else:
+            jobs = sorted(project,
                           key=lambda job: job.sp[self._sort_by],
                           reverse=self._reverse_order)
+        if self._select is not None:
+            jobs = list(filter(self._select, jobs))
```

Comment:
Why is this after `sort_by` now? With this being before the `sort_by` conditional, we only convert to a list once. Now we would do it twice if `sort_by is not None`. If we put it before, we also don't need the `list` conversion there.